cpp/src/interpolate/cubic_spline.cu

```diff
@@ -51,25 +50,24 @@ struct parallel_search {
     thrust::for_each(
       rmm::exec_policy(stream),
       thrust::make_counting_iterator<int>(0),
-      thrust::make_counting_iterator<int>(query_coords.size()),
+      thrust::make_counting_iterator<int>(t.size()),
       [p_t, p_curve_ids, p_prefixes, p_query_coords, p_result] __device__(int index) {
         int curve = p_curve_ids[index];
         int len   = p_prefixes[curve + 1] - p_prefixes[curve];
         int h     = p_prefixes[curve];
         int dh    = p_prefixes[curve] - (curve);
         // O(n) search, can do log(n) easily
-        for (int32_t i = 0; i < len; ++i) {
-          if ((p_t[h + i] + 0.0001 - p_query_coords[index]) > 0.00001) {
+        for (int32_t i = 1; i < len; ++i) {
+          if ((p_t[index] + 0.0001 < p_query_coords[h + i] + 0.00001)) {
```

Comment:
We should define these magic numbers in a constant with a descriptive name.

Reply:
Where do you recommend I put the magic numbers?

Reply:
@trxcllnt gave a good recommendation.